packages/dashboards/addon/routes/dashboards/dashboard.js

```diff
@@ -166,6 +181,30 @@ export default Route.extend({
       const dashboard = get(this, 'currentDashboard');
       get(dashboard, 'widgets').forEach(widget => widget.rollbackAttributes());
       dashboard.rollbackAttributes();
+    },
+
+    willTransition(transition) {
+      //subroute just continue
+      if (transition.targetName.startsWith('dashboards.dashboard')) {
+        return true;
+      }
+
+      const dashboard = get(this, 'currentDashboard');
+
+      const isDirty =
+        dashboard.get('hasDirtyAttributes') ||
+        dashboard.get('filters.hasDirtyAttributes') ||
+        dashboard.get('presentation.hasDirtyAttributes');
+
+      if (
+        isDirty &&
+        dashboard.get('canUserEdit') &&
+        !confirm('You have unsaved changes, are you sure you want to exit?')
```

Comment:
Is there any way to do a `Save and exit` option?

Reply:
Do we want to build a more complicated custom confirmation dialog?